addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/HeatingModel.java

```diff
@@ -131,11 +153,37 @@ public String toString() {
                 .append("nextchange", getNextchange()).toString();
     }
 
+    public static boolean inCelsiusRange(BigDecimal celsiusValue) {
+        return (TEMP_CELSIUS_MIN.compareTo(celsiusValue) <= 0) && (TEMP_CELSIUS_MAX.compareTo(celsiusValue) >= 0);
+    }
+
+    public static BigDecimal fromCelsius(BigDecimal celsiusValue) {
+        if (celsiusValue == null) {
+            return BigDecimal.ZERO;
+        } else if (TEMP_CELSIUS_MIN.compareTo(celsiusValue) == 1) {
+            return TEMP_FRITZ_MIN;
+        } else if (TEMP_CELSIUS_MAX.compareTo(celsiusValue) == -1) {
+            return TEMP_FRITZ_MAX;
+        }
+        return celsiusValue.divide(TEMP_FACTOR);
+    }
+
+    public static BigDecimal toCelsius(BigDecimal fritzValue) {
+        if (fritzValue == null) {
+            return BigDecimal.ZERO;
+        } else if (TEMP_FRITZ_ON.compareTo(fritzValue) == 0) {
+            return TEMP_CELSIUS_MAX.add(new BigDecimal("2.0"));
```

Comment:
I asked myself that question too. The used range ist from 8 - 28°C. AVM did not define if ON is 28°C or more. The device shows ON in its display. `CHANNEL_NEXTTEMP` expects type `Number`, I cannot send ON to that. So I decided to show more than 28°C. This allows me to send 28°C over openHAB to tell the FRITZ!Box that it is not ON anymore, but 28°C instead. Same for OFF.